packages/node/test/integration/setup.ts

```diff
@@ -14,7 +16,33 @@ import { MNEMONIC_PATH, Node } from "../../src";
 import { CF_PATH } from "../global-setup.jest";
 import { A_MNEMONIC, B_MNEMONIC } from "../test-constants.jest";
 
-export async function setup(
+export async function setupWithFirebaseServiceFactory(
```

Comment:
In general the concept of "service factories" feels like it shouldn't exist. It just _happens_ to be the case that Firebase can be used in a hacky way to do messaging, but in general messaging and storage are fundamentally different things and trying to introduce this kind of "service factory" wrapper concept feels like way too much overhead for this weird edge case.

Reply:
The nice property of a service factory is that it enables much easier testing as we'd need multiple instances of a service (this is only true for the store service, not the messaging service).